### PR TITLE
feat: simplify result for json

### PR DESCRIPTION
**Pull request by pllim**
Feel free to make this consistent with the rest of the package. I ran out of time at the airport. Part 1 of #41 (second part is output formal options).

```python
from scikit_hep_repo_review.processor import process, results_dict_to_json
from pathlib import Path
r = process(Path("."))
s = results_simple_dict(r)

import json
with open("zzz.json", "w") as fout:
    json.dump(s, fout)

with open("zzz.json", "r") as fin:
    dd = json.loads(json.load(fin))
```

### CODE REVIEW

**Comment by pllim** [comment]
pre-commit.ci autofix

**Comment by pllim** [comment]
Hmm weird, the autofix didn't work? Anyway, I have to log off now. I can't come back to this for a few days, so please ping me if you want me to come back and clean it up, or take it and run with it. Thanks again for this very useful tool!

File: src/scikit_hep_repo_review/processor.py [hunk 1]
```diff
@@ -105,3 +105,14 @@ def process(package: Traversable) -> dict[str, list[Result]]:
         results_dict[family] = result_list
 
     return results_dict
+
+
+def results_dict_to_json(results_dict: dict) -> str:
```
Comment (henryiii) [inline src/scikit_hep_repo_review/processor.py:110]:
```suggestion
def results_dict_to_json(results_dict: dict[str, list[Result]) -> str:
```

Though I'd probably return the object and put the json.dumps in the CLI. I can do this soon, thanks!

Comment (pllim) [reply src/scikit_hep_repo_review/processor.py:110]: Personally I would want to access the JSON stuff from API, not CLI. But however you choose to do this, I'll make it with on my end. Thanks again!

Comment (henryiii) [reply src/scikit_hep_repo_review/processor.py]: Yes, but it's easy to wrap with `json.dumps`, and then you can add indent, etc. if you want.

Comment (pllim) [reply src/scikit_hep_repo_review/processor.py]: Oh yeah, I forgot about the indent. That would be nice.

Comment (henryiii) [reply src/scikit_hep_repo_review/processor.py]: Your example actually already wrapped it in json.dumps. So the example of usage stays the same. ;)

**Comment by henryiii** [comment]
There wasn't anything it could auto fix. The only problem was with part of a type missing.

**Comment by pllim** [comment]
Thanks for the clean-up and merge!

**Pull request merged**